### ISSUE TITLE
Bulk insert with only one colum, invalid SQL is used

### ISSUE THREAD

**Issue by podhmo** [jmoiron/sqlx#709]
Bulk insert query with only one column,  generated code is something wrong. 

For example, `Named()` function is using like following.

```go
sqlx.Named("INSERT INTO vs (k) VALUES (:k)", []V{{K: "x"}, {K: "y"}})
```

Expected  SQL is here.

```
INSERT INTO vs (k) VALUES (?),(?)
```

But, generated SQL is something like this.

```
INSERT INTO vs (k) VALUES (?),(k) VALUES (?)
```

## Minimum reproducible code is here.

```go
package main

import (
	"log"

	"github.com/jmoiron/sqlx"
)

func main() {
	type V struct {
		K string `db:"k"`
		V int    `db:"v"`
	}

	{
		s, values, err := sqlx.Named("INSERT INTO vs (k) VALUES (:k)", []V{{K: "x"}, {K: "y"}})
		log.Printf("\n\tquery=%q\n\tvalues=%#+v\n\terr=%+v", s, values, err)
	}
	{
		s, values, err := sqlx.Named("INSERT INTO vs (k, v) VALUES (:k, :v)", []V{{K: "x"}, {K: "y"}})
		log.Printf("\n\tquery=%q\n\tvalues=%#+v\n\terr=%+v", s, values, err)
	}
}
``` 

the results.

```
2021/03/13 18:35:37 
	query="INSERT INTO vs (k) VALUES (?),(k) VALUES (?)"
	values=[]interface {}{"x", "y"}
	err=<nil>
2021/03/13 18:35:37 
	query="INSERT INTO vs (k, v) VALUES (?, ?),(?, ?)"
	values=[]interface {}{"x", 0, "y", 0}
	err=<nil>
```

**Comment by podhmo** [comment]
https://github.com/jmoiron/sqlx/blob/a1d5e64734233358bc4e0a54beddc18509071a95/named.go#L227
This regular expression does not support 1-column input.

**Comment by podhmo** [comment]
related to #694

**Comment by podhmo** [comment]
Ah, OK, the regular expression ending with`$` is also bad.

**Comment by podhmo** [comment]
https://github.com/jmoiron/sqlx/pull/667#issuecomment-787731678

**Comment by podhmo** [comment]
Hmm, we need to clarify the spec.
If my understanding is correct, supporting grammar is something like below  (correct me if I'm wrong).

```diff
diff --git a/named.go b/named.go
index 7f78b0d..40f2f5d 100644
--- a/named.go
+++ b/named.go
@@ -224,7 +224,10 @@ func bindStruct(bindType int, query string, arg interface{}, m *reflectx.Mapper)
        return bound, arglist, nil
 }
 
-var valueBracketReg = regexp.MustCompile(`\([^)]+(?:\s*,\s*[^)]+)*\)\s*$`)
+// expectted string is '(' <placeholder> { ',' <placeholder> }* ')'
+//     placeholder are (e.g. ?, :foo, $1)
+//     matched examples are (e.g. (?), (?, ?), (:name, :val1), ($first_name, $last_name))
+var valueBracketReg = regexp.MustCompile(`\(\s*(?:\?|[:\$][0-9a-zA-Z_]+)(?:\s*,\s*(?:\?|[:\$][0-9a-zA-Z_]+))*\s*\)`)
 ```
If this is correct, we can also handle queries such as:

```
INSERT INTO val (k) VALUES (:k) ON DUPLICATE KEY UPDATE v=v+1
```

**Comment by podhmo** [comment]
OK, cannot support this https://github.com/jmoiron/sqlx/blob/a1d5e64734233358bc4e0a54beddc18509071a95/named_test.go#L205-L207

**Comment by podhmo** [comment]
Using the last matched string is leading to solving the problem? 
(But aggressive code changes is too scary, I think)

**Comment by jmoiron** [comment]
should be fixed with v1.3.2 / #718
